pkg/protocol/engine/accounts/accountsledger/manager.go

```diff
@@ -185,6 +185,36 @@ func (m *Manager) Account(accountID iotago.AccountID, targetIndex iotago.SlotInd
 	return loadedAccount, true, nil
 }
 
+// PastAccounts loads the past accounts' data at a specific slot index.
+func (m *Manager) PastAccounts(accountIDs iotago.AccountIDs, targetIndex iotago.SlotIndex) map[iotago.AccountID]*accounts.AccountData {
```

Comment:
It is a bit ugly to expose this even if it is only used in this context. Could we relax the check on the `Account` method instead? It is nice that we enforce a strict limit there, but why do we do it anyway?

Reply:
I don't know why this check is in the `Account` method in first place. Also the method signature seems to be a bit strange to me with `account, bool, err` as the return type.

Reply:
After talking to @cyberphysic4l I think the separate method (for explicitly getting `PastAccounts`) should stay as the `Account` method has a safeguard to accidentally triggering too many disk accesses in `rollbackAccountTo`. 

> the intended use of the account manager is to load the account info when checking an arriving block. The arriving block can't be older than `latestCommittedSlot`  which means it's commitment can't be older than `latestCommittedSlot-MaxCommittableAge`